File: 2021/src/day1.rs
#[aoc(day1, part1)]
fn part1(input: &str) -> usize {
    let mut count = 0;
    let mut x = 100000;
    for l in input
        .split_ascii_whitespace()
        .map(|l| l.parse::<usize>().unwrap())
    {
        if l > x {
            count += 1;
        }
        x = l;
    }
    count
}

#[aoc(day1, part2)]
fn part2(input: &str) -> usize {
    let mut count = 0;
    let mut x = 100000;
    for l in input
        .split_ascii_whitespace()
        .map(|l| l.parse::<usize>().unwrap())
        .collect::<Vec<usize>>()
        .windows(3)
        .map(|r| r.iter().sum())
    {
        if l > x {
            count += 1;
        }
        x = l;
    }
    count
}

File: 2021/python/day1.py
#!/usr/bin/env python3

from aoc import *


def part1():
    answer = 0
    last = 1000000
    for depth in puzzle.lines(int):
        if depth > last:
            answer += 1
        last = depth
    return answer


def part2():
    answer = 0
    last = 10000000
    for depth in [sum(w) for w in windows(puzzle.lines(int), 3)]:
        if depth > last:
            answer += 1
        last = depth
    return answer


print("Part 1:", part1())
print("Part 2:", part2())
